src/socket.c

```diff
@@ -335,13 +339,47 @@ static int connSocketAddr(connection *conn, char *ip, size_t ip_len, int *port,
     return C_ERR;
 }
 
+#define NETWORK_LOOPBACK_MASK (IFF_RUNNING|IFF_LOOPBACK|IFF_UP)
 static int connSocketIsLocal(connection *conn) {
+    struct ifaddrs *ifa, *addrs;
+    struct sockaddr_in *s4;
+    struct sockaddr_in6 *s6;
+    void *in_addr;
+    char loip[NET_IP_STR_LEN];
     char cip[NET_IP_STR_LEN + 1] = { 0 };
 
     if (connSocketAddr(conn, cip, sizeof(cip) - 1, NULL, 1) == C_ERR)
         return -1;
 
-    return !strcmp(cip,"127.0.0.1") || !strcmp(cip,"::1");
+    if (getifaddrs(&addrs) != 0) return -1;
```

Comment:
i'm concerned that this API isn't fully standard and might be missing (or different) on some platforms.
can you look into it? maybe start with MacOS, FreeBSD and Solaris

Reply:
For my quick research, `getifaddrs` was introduced in FreeBSD and seems to be part of MacOS by now. It was introduced in Solaris 11 (somewhere around 2011).

The otherwise accepted way to get interface info is via `ioctl SIOCGIFCONF` (see for instance _UNIX Network Programming_).

That said, I'm starting to wonder if something like
```c
memcmp(cip, "127.", 4)
```
would also do?

Reply:
@devnexen maybe you can comment about portability as well.

Reply:
It look like `getifaddrs` [is missing](https://www.gnu.org/software/gnulib/manual/html_node/getifaddrs.html) on: AIX 5.1, HP-UX 11, IRIX 6.5, Solaris 10, Cygwin 1.5.x, mingw, MSVC 14, Android 6.0. I don't know how many of those (if any) Redis aims to support.

But IMHO it sounds like just checking for the "127." prefix should work fine?

Reply:
> i'm concerned that this API isn't fully standard and might be missing (or different) on some platforms. can you look into it? maybe start with MacOS, FreeBSD and Solaris

it is safe from BSD point of view.